External/Plugins/CodeRefactor/Commands/RenameFile.cs

```diff
@@ -48,17 +48,19 @@ protected override void ExecutionImplementation()
                     return;
                 }
             }
+
+            newPath = Path.Combine(Path.GetDirectoryName(oldPath), newPath);
+
             // refactor failed or was refused
             if (Path.GetFileName(oldPath).Equals(newPath, StringComparison.OrdinalIgnoreCase))
             {
                 // name casing changed
                 string tmpPath = oldPath + "$renaming$";
                 File.Move(oldPath, tmpPath);
-                oldPath = tmpPath;
+                File.Move(tmpPath, newPath);
+                DocumentManager.MoveDocuments(oldPath, newPath);
             }
-            if (!Path.IsPathRooted(newPath)) newPath = Path.Combine(Path.GetDirectoryName(oldPath), newPath);
```

Comment:
Why did you remove this one? I should check my notes... but IIRC it was this exact one that back in the day I also thought it was unneeded and didn't find any valid reason for it but left it just in case...